core/src/main/java/org/apache/spark/util/collection/unsafe/sort/UnsafeInMemorySorter.java

```diff
@@ -216,7 +216,7 @@ public void expandPointerArray(LongArray newArray) {
     if (array != null) {
       if (newArray.size() < array.size()) {
         // checkstyle.off: RegexpSinglelineJava
```

Comment:
Is it needed, BTW?

Reply:
Let me take a look

Reply:
The current is necessary because of the following inspection rule:

https://github.com/apache/spark/blob/3b0ac45391708642ce6a1779e3c234bab0e40b66/dev/checkstyle.xml#L171-L174

If removed, running `dev/lint-java` will result in the following error: 

```
[ERROR] src/main/java/org/apache/spark/util/collection/unsafe/sort/UnsafeInMemorySorter.java:[218] (regexp) RegexpSinglelineJava: Avoid throwing error in application code.
```